src/libraries/System.IO.FileSystem/src/System/IO/FileStatus.OSX.cs

```diff
@@ -0,0 +1,70 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+
+namespace System.IO
+{
+    internal partial struct FileStatus
+    {
+        internal unsafe void SetCreationTime(string path, DateTimeOffset time)
+        {
+            Interop.Sys.TimeSpec timeSpec = default;
+
+            long seconds = time.ToUnixTimeSeconds();
+
+            const long TicksPerMillisecond = 10000;
+            const long TicksPerSecond = TicksPerMillisecond * 1000;
+            long nanoseconds = (time.UtcDateTime.Ticks - DateTimeOffset.UnixEpoch.Ticks - seconds * TicksPerSecond) * NanosecondsPerTick;
+
+            timeSpec.TvSec = seconds;
+            timeSpec.TvNsec = nanoseconds;
+
+            Interop.libc.AttrList attrList = default;
+            attrList.bitmapCount = Interop.libc.AttrList.ATTR_BIT_MAP_COUNT;
+            attrList.reserved = 0;
+            attrList.commonAttr = Interop.libc.AttrList.ATTR_CMN_CRTIME;
+            attrList.dirAttr = 0;
+            attrList.fileAttr = 0;
+            attrList.forkAttr = 0;
+            attrList.volAttr = 0;
```

Comment:
Since you set the `attrList` variable to `default`, all the `uint` and `ushort` fields are automatically zero'd, so you don't need to do it yourself:
```suggestion
            Interop.libc.AttrList attrList = default;
            attrList.bitmapCount = Interop.libc.AttrList.ATTR_BIT_MAP_COUNT;
            attrList.commonAttr = Interop.libc.AttrList.ATTR_CMN_CRTIME;
```

Reply:
I've added this now.